vip-scanner/checks/VIPRestrictedVariablesCheck.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+class VIPRestrictedVariablesCheck extends BaseCheck
+{
+	function check( $files ) {
+		$result = true;
+
+		$checks = array(
+			'((?<![\\\'\"])\$\$[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*?(?![\\\'\"])|(?<![\\\'\"])\$\{(?:.*)[\}](?![\\\'\"]))' => array( "level" => "Warning", "note" => "Possible variable variables" ),
```

Comment:
>There are some other interesting possibilities we can try to account for:
>```php
${"foo$one"} = 'foo'; 
${'foo' . ONE} = 'bar'; 
${'foo' . one()} = 'baz';
```
>http://php.net/manual/en/language.variables.variable.php#75669

All of these should be caught with `\$\{(?:.*)[\}]` since it shouldn't care what's inside the complex syntax brackets, including nested complex syntax statements.